Objects/unicodeobject.c

```diff
@@ -14784,14 +14784,15 @@ PyDoc_STRVAR(length_hint_doc, "Private method returning an estimate of len(list(
 static PyObject *
 unicodeiter_reduce(unicodeiterobject *it, PyObject *Py_UNUSED(ignored))
 {
+    PyObject *iter = _PyEval_GetBuiltin(&_Py_ID(iter));
+
```

Comment:
Any reason not to include the same comment here as in the other cases?

Reply:
Not sure, I think I originally thought this wasn't reproducible in standard conditions since strings are immutable, but it very much is since it's just a string iterator being consumed. 

Added similar comment in [93854e1](https://github.com/python/cpython/pull/101769/commits/93854e172e0b4c7cb3b999bdec6c97da74853a90)